tls/s2n_quic_support.c

```diff
@@ -90,24 +90,46 @@ int s2n_connection_set_secret_callback(struct s2n_connection *conn, s2n_secret_c
     return S2N_SUCCESS;
 }
 
+/* Currently we need an API that quic can call to process post-handshake messages. Ideally
+ * we could re-use the s2n_recv API but that function needs to be refactored to support quic.
+ * For now we just call this API.
+ */
+int s2n_connection_process_post_handshake_message(struct s2n_connection *conn)
+{
+    POSIX_ENSURE_REF(conn);
+
+    uint8_t message_type = 0;
+    struct s2n_blob b = { 0 };
+    struct s2n_stuffer header = { 0 };
+    POSIX_GUARD(s2n_blob_init(&b, conn->post_handshake.header_in, sizeof(conn->post_handshake.header_in)));
+    POSIX_GUARD(s2n_stuffer_init(&header, &b));
+    POSIX_GUARD(s2n_stuffer_growable_alloc(&conn->post_handshake.in, 0));
```

Comment:
Are you sure about this?
* header: if you initialize the stuffer every time, how is it tracking progress across calls? What happens to the header on the second call if the first blocks?
* message: What if conn->post_handshake.in is already alloc'ed?

I think you definitely need some blocking / fragmentation tests.

Reply:
Added fragmentation tests, not sure about blocking tests, I don't think this function could block given how quic would call it?

Reply:
Your fragmentation tests are blocking tests :) When your new method receives a fragment, it blocks, then can be successfully called again to process the next fragment.